openfasoc/generators/ldo-gen/simulations/templates/ldoInst_ngspice.sp

```diff
@@ -44,34 +44,36 @@ vstd6 std_pt_in_cnt[6] 0 dc 0
 vstd7 std_pt_in_cnt[7] 0 dc 0
 vstd8 std_pt_in_cnt[8] 0 dc 0
 
-R1 VREG VSS 3.6k
+R1 VREG VSS @Res_Value
 *Resistance 3600 --> 0.5 mA for 1.8 V reference voltage. R to be adjusted according to Iload and output voltage.
 
 *R0 VREG 0 R='TIME > 500u ? 720 : 3600' ;If want to do a transient simulation where I load is changing in the middle of a transient simulation. Example: Here it is changing from 720 ohm (2.5mA) to 3600 ohm (0.5 mA) at t = 1000us
 
-C1 VREG VSS 30n
+C1 VREG VSS @Cap_Value
 
 *.options savecurrents
+.option wnflag=1
 .options rshunt=1e11
 .ic v(VREG) = 0 v(clk)=0 v(reset)=3.3
 *Analysis
 .temp 25
-.tran 100n 300u
+.tran @sim_step @sim_time
 
 .probe V(VREG) v(VREF) v(cmp_out) v(clk) i(R1)
 .control
 run
 
-set hcopydevtype = svg
-set svg_intopts = ( 2560 1440 30 0 1 2 0 )
-setcs svg_stropts = ( white Arial Arial )
-set color1 = black
-set color2 = red
-hardcopy vregPlot.svg v(VREG) title 'LDO VREG startup transients with 1.8V VREF'
-hardcopy currentPlot.svg I(R1) title 'LDO load current startup transients with 1.8V VREF'
+*set hcopydevtype = svg
+*set svg_intopts = ( 2560 1440 30 0 1 2 0 )
+*setcs svg_stropts = ( white Arial Arial )
+*set color1 = black
+*set color2 = red
+*hardcopy vregPlot.svg v(VREG) title 'LDO VREG startup transients with 1.8V VREF'
+*hardcopy currentPlot.svg I(R1) title 'LDO load current startup transients with 1.8V VREF'
 *hardcopy cmp_out_clk_plot.ps cmp_out clk
 
-write output.raw v(VREG) v(VREF) v(cmp_out) v(clk) (("ctrl_out[0]"+2*"ctrl_out[1]"+4*"ctrl_out[2]"+8*"ctrl_out[3]"+16*"ctrl_out[4]"+32*"ctrl_out[5]"+ 64*"ctrl_out[6]"+128*"ctrl_out[7]"+ 256*"ctrl_out[8]")/3.3) ;Last variable is for the counter indicating number of switches that are turning ON (ctrl_word_cnt)
+set filetype=binary
+write @output_raw v(VREG) v(VREF) v(cmp_out) v(clk)
```

Comment:
@lulu9312 and @ayushman22git Do we want a one single testbench where we run all the sims or should we have single unit test and then one single test bench with all the blocks connected?

Reply:
I think we should have seperate testbench for seperate blocks so that when we are automating the the sim runs we can handle them accordingly. What do you think? @ayushman22git

Reply:
Agreed. Having separate files for different blocks will help reduce simulation time when one wants to optimize individual blocks and needs to run multiple simulations for that.

Reply:
This sounds interesting. Not just with the run time, if a block is modified then the simulation metrics can also be calculated to compare its effect on the overall block for various possible configs.

Reply:
Also, just out of curiosity, by when the simulations of this generator can be tried by others? I just want to see what changes need to be done to CI in order to add sims (maybe in general) as this simulation looks promising and trustworthy since you all are working on it